src/Microsoft.QualityGuidelines.Analyzers/Core/MarkMembersAsStatic.Fixer.cs

```diff
@@ -22,9 +29,33 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await Task.Run(() => { }).ConfigureAwait(false);
-            throw new NotImplementedException();
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            context.RegisterCodeFix(new MarkMembersAsStaticAction(
+                    string.Format(MicrosoftQualityGuidelinesAnalyzersResources.MarkMembersAsStaticCodeFix, symbol.Name),
+                    async ct => await RemoveStaticModifier(context.Document, root, node, ct).ConfigureAwait(false)
+                ), context.Diagnostics.Single());
```

Comment:
single might be too restrictive since you can't control another analyzer issuing same diagnostic on same location. analyzer and fixer model is loosely coupled model meaning different analyzer can issue same diagnostics that one fixer can work on both diagnostics.